android/walletlib/src/main/java/com/solana/mobilewalletadapter/walletlib/authorization/AuthRepository.java

```diff
@@ -99,6 +100,7 @@ private SQLiteDatabase ensureStarted() {
             }
 
             mAuthDb = new AuthDatabase(mContext, mAuthIssuerConfig);
+            mIdentityRecordDao = new IdentityRecordDao(mAuthDb.getWritableDatabase());
```

Comment:
`getWritableDatabase` shouldn't be called from the application main thread: https://developer.android.com/reference/android/database/sqlite/SQLiteOpenHelper#getWritableDatabase()

`ensureDatabase` was used to lazily defer this to the caller thread (which was always done in an IO context, not Main context)